crates/gpui/src/style.rs

```diff
@@ -46,7 +46,8 @@ impl ObjectFit {
         bounds: Bounds<Pixels>,
         image_size: Size<DevicePixels>,
     ) -> Bounds<Pixels> {
-        let image_size = image_size.map(|dimension| Pixels::from(u32::from(dimension)));
+        // NB: this reinterprets physical pixels as logical pixels without scaling them properly
+        let image_size = image_size.map(|dimension| px(dimension.unquantize().0));
```

Comment:
This PR should include the state transitions between `PhysicalPixel`s and `Pixel`s, and these scaling related bugs should be shown explicitly like `dimension.unquantize().to_logical(1)`.

Reply:
I'll add `to_logical` and try to use it everywhere where it applies.

There are also quite a few existing conversion functions: `Point::<Pixels>::scale`, `Pixels::scale`, `Size::<PhysicalPixels<i32>>::to_pixels`, `Size::<Pixels>::to_device_pixels`, `Bounds::<Pixels>::{scale,to_device_pixels}`, `Bounds::<DevicePixels>::to_pixels`. I think ideally we'd rename them to `to_logical` /`to_physical`, should I do this in this PR or leave it as a follow up similarly to `ScaledPixels`/`DevicePixels` aliases?

Reply:
It depends on how invasive / difficult the change is. If you're running into a lot of merge conflicts or touch a lot of files, it might be better to do in a follow up for all of our sanity :)

However, I don't like to leave a job half done, so I'd prefer to have it if we can. I'll defer to your judgement in this case.